content/en/docs/reference/using-api/server-side-apply.md

```diff
@@ -313,8 +313,7 @@ read-modify-write and/or patch are the following:
 * the applied object must contain all the fields that the controller cares about.
 * there is no way to remove fields that haven't been applied by the controller
   before (controller can still send a PATCH/UPDATE for these use-cases).
-* the object doesn't have to be read beforehand, `resourceVersion` doesn't have
-  to be specified.
+* the object doesn't have to be read beforehand, `resourceVersion` must not be specified.
```

Comment:
since docs say that object doesn't have to be read beforehand maybe it should also mention that this doesn't guarantee that version conflict won't happen?

Reply:
if a controller made a decision about the content to apply based on reading the current state of the object, then adding a resourceVersion as a precondition is very reasonable (and in fact the only way to be sure other unowned fields didn't change in the meantime, possibly invalidating the computed content to apply)

Reply:
I think it's OK to explain the two cases:
- using server-side apply to avoid needing to care about conflicts. For example, if you're a controller that labels any non-system namespace with `foo: bar`, you can apply that label regardless of the existing object
- using server-side apply to assert ownership of a field, provided prerequisites are met. For example, if you're a controller that annotates Deployments with a verdict on the container images mentioned in the `spec`, but only if the images are from an in-cluster container registry.

Reply:
(if someone edits the Deployment whilst you're not looking, you would **apply** an incorrect verdict - a classic lost update)